sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCTableCatalogSuite.scala

```diff
@@ -77,8 +77,8 @@ class JDBCTableCatalogSuite extends QueryTest with SharedSparkSession {
     sql("DROP TABLE h2.test.to_drop")
     checkAnswer(sql("SHOW TABLES IN h2.test"), Seq(Row("test", "people")))
     Seq(
-      "h2.test.not_existing_table" -> "Table test.not_existing_table not found",
-      "h2.bad_test.not_existing_table" -> "Table bad_test.not_existing_table not found"
+      "h2.test.not_existing_table" -> "Table h2.test.not_existing_table not found",
+      "h2.bad_test.not_existing_table" -> "Table h2.bad_test.not_existing_table not found"
```

Comment:
Now that table is resolved before `drop table` runs, when the table is not found, it will always print the original identifier.